src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -96,6 +96,19 @@ private static void PingResultValidator(PingReply pingReply, IPAddress localIpAd
             AssertExtensions.Throws<ArgumentException>("buffer", () => { p.Send(TestSettings.LocalHost, 1, new byte[65501]); });
         }
 
+        [Fact]
+        public void SendPingWithIPAddress()
+        {
+            IPAddress localIpAddress = TestSettings.GetLocalIPAddress().GetAwaiter().GetResult();
```

Comment:
Just out of curiosity, why `.GetAwaiter().GetResult()` vs just `.Result` here?

Reply:
This pattern will throw the exception without it being wrapped in AggregateException.

However, we shouldn't be using this at all.

This test should be async Task and the code should be like this:

```C#
IPAddress localIpAddress = await TestSettings.GetLocalIPAddress();
```

And if this TestSettings method is async it should be named appropriately, i.e. "TestSettings.GetLocalIPAddressAsync()"

Reply:
This also is the way it was previously.  Most of the code in this PR is just me reorganising code to maximise what could be used on the sync and async paths.

I think reason why this is the way it is though was because the original author didn't want to expose "async" methods because they wanted exceptions to be thrown on the calling thread and not be wrapped in AggregateException.  If I follow the suggestion then I'll end up having to change the signature of the top level method and this could be seen to be a breaking change due to the change in the way exceptions are thrown.

Reply:
Ah, sorry - just noticed this was a test.  There is code in one of the actual Ping*.cs classes that looks the same.  No problem changing this

Reply:
I think the fix you made for this comment wasn't quite right. The tests should still be async, even if they tests the synchronous ping implementation. That should allow you to use the async implementation of `GetLocalIPAddress`, and you can avoid some code changes there.

Reply:
I'm happy to cut back on this but if you don't mind I'd like to wait to see if @davidsh has thoughts on this first, just to save any toing-and-froing.

The review obviously factored in code that already existed - David said that `GetLocalIPAddress` should be called `GetLocalIPAddressAsync`.  Since I was already fixing one existing problem I thought I may as well fix another related one, i.e. there were lots of async methods that really shouldn't have been async.  While we might not be worried about the tiny performance hit with the tests I think the bigger potential danger is that having those methods being async means that it wouldn't be difficult to fat-finger with intellisense and end up accidentally testing an async Ping method rather than the intended sync one - at least now the compiler will prevent this.